### PR TITLE
feat: add optimism

### PR DESCRIPTION
**Pull request by kaladinlight**
## Description

Add base support for Optimism

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

## Risk

Low - additive

## Testing

- Send
- Receive
- Balance
- Transaction History

plz also check that anywhere we display asset icons are showing correctly with colors and little accented network icons, e.g. asset list, accounts page, dashboard, asset pages etc.

native wallet specifically. can spot check metamask, but was having trouble sending at the moment. the rest _should_ work

### Engineering

:point_up: 

### Operations

:point_up: 

## Screenshots (if applicable)

image
image
image
image
image
image

### CODE REVIEW

File: .env.app [hunk 1]
```diff
@@ -18,6 +18,8 @@ REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL=https://api.ethereum.shapeshift.com
 REACT_APP_UNCHAINED_ETHEREUM_WS_URL=wss://api.ethereum.shapeshift.com
 REACT_APP_UNCHAINED_AVALANCHE_HTTP_URL=https://api.avalanche.shapeshift.com
 REACT_APP_UNCHAINED_AVALANCHE_WS_URL=wss://api.avalanche.shapeshift.com
+REACT_APP_UNCHAINED_OPTIMISM_HTTP_URL=https://dev-api.optimism.shapeshift.com
```
Comment (0xApotheosis) [inline .env.app:21]: If we are using the same URL for all (or most environments) do we just want to use `.env.base`?

Comment (kaladinlight) [reply .env.app:21]: it won't be that way long term, just need to stand up prod coinstacks still

File: src/components/AssetIcon.tsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import { Avatar, Circle, useColorModeValue, useMultiStyleConfig } from '@chakra-ui/react'
+import type { Asset } from '@shapeshiftoss/asset-service'
 import type { AssetId } from '@shapeshiftoss/caip'
 import { selectAssetById, selectFeeAssetById } from 'state/slices/selectors'
 import { useAppSelector } from 'state/store'
 
 import { FoxIcon } from './Icons/FoxIcon'
 
 type AssetIconProps = {
+  asset?: Asset
```
Comment (gomesalexandre) [inline src/components/AssetIcon.tsx:11]: Fine for now but we might want to make it a discriminated union in the future

File: src/plugins/optimism/index.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                )
+
+                const ws = new unchained.ws.Client<unchained.optimism.Tx>(
+                  getConfig().REACT_APP_UNCHAINED_OPTIMISM_WS_URL,
+                )
+
+                return new optimism.ChainAdapter({
+                  providers: { http, ws },
+                  rpcUrl: getConfig().REACT_APP_OPTIMISM_NODE_URL,
+                }) as unknown as ChainAdapter<ChainId> // FIXME: this is silly
```
Comment (gomesalexandre) [inline src/plugins/optimism/index.tsx:35]: I recognize this comment from old implementations but I reckon we should be able to type this as `ChainAdapter<KnownChainIds.OptimismMainnet>` now?

**Review by gomesalexandre — COMMENTED** [review]
Conceptual stamp pending testing

**Review by gomesalexandre — APPROVED** [review]
LGTM, tested:

#### /accounts account

image

#### /account account

image

#### Receive
 
image

#### Sends

image
image
image

Checked for regressions WRT the usage of `feeAsset.networkIcon` and `feeAsset.networkName` and confirmed we are good here

Tested fiat ramps as a sanity check and confirmed that as expected, optimism isn't there for now. To be added lib, since Mt Pelerin supports it https://www.optimism.io/apps/fiat-on-ramp

**Pull request merged**